eel/src/recovery.rs

```diff
@@ -10,16 +21,120 @@ pub fn recover_lightning_node(
     remote_storage: Box<dyn RemoteStorage>,
 ) -> Result<()> {
     let encryption_key = key_derivation::derive_persistence_encryption_key(&seed).unwrap();
-    let _persister = Arc::new(StoragePersister::new(
+    let storage = Arc::new(StoragePersister::new(
         remote_storage,
-        local_persistence_path,
+        local_persistence_path.clone(),
         encryption_key,
     ));
 
-    // TODO:
-    //      * check if there's a local installation, if yes, return InvalidInput
-    //      * check if there's files in the backend, if not, return RuntimeError with code NonExistingWallet
-    //      * recover
+    if has_local_install(&local_persistence_path) {
+        return Err(invalid_input(
+            "Invalid local persistence path: an existing wallet installation was found!",
+        ));
+    }
+
+    fs::create_dir_all(&local_persistence_path)
+        .map_to_invalid_input("Invalid local persistence path: failed to create directory")?;
+
+    // Fetch and persist ChannelManager
+    let remote_channel_manager = storage.fetch_remote_channel_manager_serialized()?;
+    fs::write(
+        get_local_channel_manager_path(&local_persistence_path),
+        remote_channel_manager,
+    )
+    .map_to_permanent_failure(
+        "Failed to locally persist the ChannelManager recovered from remote storage",
+    )?;
+
+    // Fetch and persist ChannelMonitors
+    let mut seed_first_half = [0u8; 32];
+    seed_first_half.copy_from_slice(&seed[..32]);
+    let keys_manager = Arc::new(init_keys_manager(&seed_first_half)?);
+
+    let remote_channel_monitors =
+        storage.fetch_remote_channel_monitors(&*keys_manager, &*keys_manager)?;
+    info!(
+        "Fetched {} channel monitors from remote storage during recovery procedure",
+        remote_channel_monitors.len()
+    );
+
+    let persister = FilesystemPersister::new(local_persistence_path);
+
+    for (_, monitor) in remote_channel_monitors {
+        persist_channel_monitor(&persister, monitor.get_funding_txo().0, &monitor)?;
+    }
 
     Ok(())
 }
+
+fn has_local_install(local_persistence_path: &str) -> bool {
+    // ChannelManager
+    let channel_manager_path = get_local_channel_manager_path(local_persistence_path);
+    if fs::File::open(channel_manager_path).is_ok() {
+        return true;
+    }
+    // ChannelMonitors
+    let channel_monitors_dir_path = get_local_channel_monitors_dir_path(local_persistence_path);
+    if let Ok(mut dir_entries) = fs::read_dir(channel_monitors_dir_path) {
+        if dir_entries.next().is_some() {
+            return true;
+        }
+    }
+    false
+}
```

Comment:
Get rid off the comments and make the code more readable by making functions single-responsibility:

```suggestion
fn has_local_install(local_persistence_path: &str) -> bool {
    has_local_channel_manager(local_persistence_path)
        || has_local_channel_monitors(local_persistence_path)
}

fn has_local_channel_manager(local_persistence_path: &str) -> bool {
    fs::File::open(get_local_channel_manager_path(local_persistence_path)).is_ok()
}

fn has_local_channel_monitors(local_persistence_path: &str) -> bool {
    let channel_monitors_dir_path = get_local_channel_monitors_dir_path(local_persistence_path);
    match fs::read_dir(channel_monitors_dir_path) {
        Ok(mut dir_entries) => dir_entries.next().is_some(),
        Err(_) => false,
    }
}
```